KitePrintSDK/src/main/java/ly/kite/analytics/Analytics.java

```diff
@@ -356,6 +358,36 @@ private Analytics( Context context )
 
   /*****************************************************
    *
+   * Gets the integrating applications analytics event
+   * callback if any, else returns null. The event
+   * callback class is registered by the integrating
+   * application as a string resource
+   * (R.string.analytics_event_callback_class_name) which we
+   * instantiate via reflection & cache for future use
+   *
+   *****************************************************/
+  private IAnalyticsEventCallback getEventCallback()
+    {
+    if ( mCachedEventCallback == null )
+      {
+      // Load registered analytics event callback (if any) via reflection
+      String callbackClassName = mContext.getString(R.string.analytics_event_callback_class_name);
+      try
+        {
+        Class<?> clazz = Class.forName( callbackClassName );
+        Constructor<?> ctor = clazz.getConstructor( Context.class );
+        mCachedEventCallback = (IAnalyticsEventCallback) ctor.newInstance( new Object[] { mContext } );
+      } catch ( Exception ex )
+        {
+        mCachedEventCallback = null;
```

Comment:
Good point, will do that now. I added the Null callback as a last minute thing just to check it all works as expected and entirely forgot I did intend to remove those null checks.